ui/app/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -0,0 +1,124 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import PermissionsConnectHeader from './permissions-connect-header'
+import PermissionsConnectFooter from './permissions-connect-footer'
+import ChooseAccount from './choose-account'
+import PermissionPageContainer from '../../components/app/permission-page-container'
+
+export default class PermissionConnect extends Component {
+  static propTypes = {
+    approvePermissionsRequest: PropTypes.func.isRequired,
+    rejectPermissionsRequest: PropTypes.func.isRequired,
+    currentMetaMaskTabOpenerId: PropTypes.number,
+    getOpenMetaMaskTabs: PropTypes.func.isRequired,
+    getCurrentWindowTab: PropTypes.func.isRequired,
+    accounts: PropTypes.array.isRequired,
+    originName: PropTypes.string,
+    showNewAccountModal: PropTypes.func.isRequired,
+    newAccountNumber: PropTypes.number.isRequired,
+    nativeCurrency: PropTypes.string,
+    permissionsRequest: PropTypes.object,
+  }
+
+  static defaultProps = {
+    originName: '',
+    nativeCurrency: '',
+    permissionsRequest: {},
+    currentMetaMaskTabOpenerId: null,
+  }
+
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  state = {
+    page: 1,
+    selectedAccountAddress: '',
+    permissionAccepted: null,
+  }
+
+  selectAccount (address) {
+    this.setState({
+      page: 2,
+      selectedAccountAddress: address,
+    })
+  }
+
+  redirectFlow (accepted) {
+    const { currentMetaMaskTabOpenerId } = this.props
+    this.setState({
+      page: null,
+      permissionAccepted: accepted,
+    })
+    setTimeout(() => {
+      global.platform.currentTab()
+        .then(({ id: currentTabId }) => {
+          global.platform.switchToTab(currentMetaMaskTabOpenerId, () => {
+            global.platform.closeTab(currentTabId)
+          })
+        })
+    }, 2000)
+  }
+
+  componentDidMount () {
+    const {
+      getOpenMetaMaskTabs,
+      getCurrentWindowTab,
+    } = this.props
+    getCurrentWindowTab()
+    getOpenMetaMaskTabs()
+  }
+
+  render () {
+    const {
+      approvePermissionsRequest,
+      rejectPermissionsRequest,
+      accounts,
+      originName,
+      showNewAccountModal,
+      newAccountNumber,
+      nativeCurrency,
+      permissionsRequest,
+    } = this.props
+    const { page, selectedAccountAddress, permissionAccepted } = this.state
+
+    return (
+      <div className="permissions-connect">
+        { page !== null
+          ? <PermissionsConnectHeader page={page} />
+          : null
+        }
+        { page === 1
+          ? <ChooseAccount
+            accounts={accounts}
+            originName={originName}
+            nativeCurrency={nativeCurrency}
+            selectAccount={(address) => this.selectAccount(address)}
+            selectNewAccountViaModal={() => {
```

Comment:
I have not QA'd `selectNewAccountViaModal`... this should either be e2e tested or verified via a video recorded QA'd

Reply:
- [ ] make sure to QA selectNewAccountViaModal

Reply:
QA'd: works but not styled.

Reply:
- [ ] Style new account modal opened here